clang/lib/Sema/SemaDecl.cpp

```diff
@@ -11576,22 +11584,6 @@ static bool CheckTargetCausesMultiVersioning(Sema &S, FunctionDecl *OldFD,
     }
   }
 
-  if (NewTVA) {
```

Comment:
This is checking things like
```
//expected-note@+1 {{previous declaration is here}}
void __attribute__((target_version("bti+flagm2"))) one(void) {}
//expected-error@+1 {{multiversioned function redeclarations require identical target attributes}}
void __attribute__((target_version("flagm2+bti"))) one(void) {}
 ```
 (example taken from clang/test/Sema/attr-target-version.c)
 
 At this point we know that the new declaration is triggering multiversioning therefore the old was the (implicit or explicit) default.

Reply:
If you are reasoning based on that, it might be good to add an assert checking it:
```
assert(!OldTVA || OldTVA->isDefaultVersion());
```

Reply:
The assertion should be conditional when NewTVA is not null because the function is used for the target attribute as well. Do we still want the assertion?